### PR TITLE
fix(DB/SAI): Shay Leafrunner

### PR DESCRIPTION
**Pull request by The-GhostRider**
## Changes Proposed:
-  Fix NPC: [Shay Leafrunner](https://wowgaming.altervista.org/aowow/?npc=7774)

## Issues Addressed:

- Closes https://github.com/chromiecraft/chromiecraft/issues/2186
- Closes https://github.com/azerothcore/azerothcore-wotlk/issues/16453

## Tests Performed:

- Tested in game and working
- Tested on Windows 10 (64)
- **Video Before my Fix with AI in C++:** [AI C++ npc_shay_leafrunner bugs](https://youtu.be/Oz1DeeJ2KCQ)
- **Video Test after my SAI Fix:** [Quest - Wandering Shay](https://youtu.be/GgBu035zGWw)

**[Note]:**

_Generally my PRs don't need to be reviewed as I always attach Videos that demonstrate how they work, so once my PR pass the Bot reviews, just watch my Videos, after which they can be merged.
Anyway, if you wish test my work in any case, feel free to do so, I do so just to save you time and optimize "Bureaucracy" (let's say).._

## How to Test the Changes:

1.
2.
3.

## Known Issues and TODO List:

- [ ]
- [ ]

## How to Test AzerothCore PRs
 
When a PR is ready to be tested, it will be marked as **[WAITING TO BE TESTED]**.

You can help by testing PRs and writing your feedback here on the PR's page on GitHub. Follow the instructions here:

http://www.azerothcore.org/wiki/How-to-test-a-PR

**REMEMBER**: when testing a PR that changes something **generic** (i.e. a part of code that handles more than one specific thing), the tester should not only check that the PR does its job (e.g. fixing spell XXX) but **especially** check that the PR does not cause any regression (i.e. introducing new bugs).

**For example**: if a PR fixes spell X by changing a part of code that handles spells X, Y, and Z, we should not only test X, but **we should test Y and Z as well**.

### CODE REVIEW

**Comment by Kitzunu** [comment]
in this case your title should be `fix(DB/Creature)`

We use the "base" class, such as Creature/Instance/GameObject/Spell etc. in the second part, never table names 👌

**Comment by The-GhostRider** [comment]
> in this case your title should be `fix(DB/Creature)`
> 
> We use the "base" class, such as Creature/Instance/GameObject/Spell etc. in the second part, never table names 👌

I will try to follow your instructions, in case I make another mistake, I apologize in advance! 😅

File: data/sql/updates/pending_db_world/rev_1685747385631953200.sql [hunk 1]
```diff
@@ -0,0 +1,41 @@
...
+
+DELETE FROM `smart_scripts` WHERE (`source_type` = 9 AND `entryorguid` = 777403);
+INSERT INTO `smart_scripts` (`entryorguid`, `source_type`, `id`, `link`, `event_type`, `event_phase_mask`, `event_chance`, `event_flags`, `event_param1`, `event_param2`, `event_param3`, `event_param4`, `event_param5`, `action_type`, `action_param1`, `action_param2`, `action_param3`, `action_param4`, `action_param5`, `action_param6`, `target_type`, `target_param1`, `target_param2`, `target_param3`, `target_param4`, `target_x`, `target_y`, `target_z`, `target_o`, `comment`) VALUES
+(777403, 9, 0, 0, 0, 0, 100, 0, 0, 0, 0, 0, 0, 6, 2845, 0, 0, 0, 0, 0, 18, 10, 0, 0, 0, 0, 0, 0, 0, 'Shay Leafrunner - Actionlist - Fail Quest \'Wandering Shay\''),
+(777403, 9, 1, 0, 0, 0, 100, 0, 0, 0, 0, 0, 0, 18, 768, 0, 0, 0, 0, 0, 1, 0, 0, 0, 0, 0, 0, 0, 0, 'Shay Leafrunner - Actionlist - Set Flags Immune To Players & Immune To NPC\'s'),
+(777403, 9, 2, 0, 0, 0, 100, 0, 0, 0, 0, 0, 0, 74, 61, 0, 0, 0, 0, 0, 1, 0, 0, 0, 0, 0, 0, 0, 0, 'Shay Leafrunner - Actionlist - Remove Timed Event 61');
```
Comment (Gultask) [inline data/sql/updates/pending_db_world/rev_1685747385631953200.sql:41]:
```suggestion
DELETE FROM `smart_scripts` WHERE (`source_type` = 9 AND `entryorguid` IN (777400, 777401, 777402, 777403));
INSERT INTO `smart_scripts` (`entryorguid`, `source_type`, `id`, `link`, `event_type`, `event_phase_mask`, `event_chance`, `event_flags`, `event_param1`, `event_param2`, `event_param3`, `event_param4`, `event_param5`, `action_type`, `action_param1`, `action_param2`, `action_param3`, `action_param4`, `action_param5`, `action_param6`, `target_type`, `target_param1`, `target_param2`, `target_param3`, `target_param4`, `target_x`, `target_y`, `target_z`, `target_o`, `comment`) VALUES
(777400, 9, 0, 0, 0, 0, 100, 0, 0, 0, 0, 0, 0, 19, 768, 0, 0, 0, 0, 0, 1, 0, 0, 0, 0, 0, 0, 0, 0, 'Shay Leafrunner - Actionlist - Remove Flags Immune To Players & Immune To NPC\'s'),
(777400, 9, 1, 0, 0, 0, 100, 0, 0, 0, 0, 0, 0, 29, 2, 3, 7765, 0, 0, 0, 7, 0, 0, 0, 0, 0, 0, 0, 0, 'Shay Leafrunner - Actionlist - Start Follow Invoker'),
(777400, 9, 2, 0, 0, 0, 100, 0, 0, 0, 0, 0, 0, 1, 0, 0, 0, 0, 0, 0, 1, 0, 0, 0, 0, 0, 0, 0, 0, 'Shay Leafrunner - Actionlist - Say Line 0'),
(777400, 9, 3, 0, 0, 0, 100, 0, 0, 0, 0, 0, 0, 45, 1, 1, 0, 0, 0, 0, 1, 0, 0, 0, 0, 0, 0, 0, 0, 'Shay Leafrunner - Actionlist - Set Data 1 1'),
(777401, 9, 0, 0, 0, 0, 100, 0, 0, 0, 0, 0, 0, 29, 2, 3, 7765, 0, 0, 0, 7, 0, 0, 0, 0, 0, 0, 0, 0, 'Shay Leafrunner - Actionlist - Start Follow Invoker'),
(777401, 9, 1, 0, 0, 0, 100, 0, 0, 0, 0, 0, 0, 1, 1, 0, 0, 0, 0, 0, 1, 0, 0, 0, 0, 0, 0, 0, 0, 'Shay Leafrunner - Actionlist - Say Line 1'),
(777402, 9, 0, 0, 0, 0, 100, 0, 0, 0, 0, 0, 0, 15, 2845, 0, 0, 0, 0, 0, 18, 10, 0, 0, 0, 0, 0, 0, 0, 'Shay Leafrunner - Actionlist - Quest Credit \'Wandering Shay\''),
(777402, 9, 1, 0, 0, 0, 100, 0, 0, 0, 0, 0, 0, 1, 4, 0, 0, 0, 0, 0, 1, 0, 0, 0, 0, 0, 0, 0, 0, 'Shay Leafrunner - Actionlist - Say Line 4'),
(777403, 9, 0, 0, 0, 0, 100, 0, 0, 0, 0, 0, 0, 6, 2845, 0, 0, 0, 0, 0, 18, 10, 0, 0, 0, 0, 0, 0, 0, 'Shay Leafrunner - Actionlist - Fail Quest \'Wandering Shay\''),
(777403, 9, 1, 0, 0, 0, 100, 0, 0, 0, 0, 0, 0, 18, 768, 0, 0, 0, 0, 0, 1, 0, 0, 0, 0, 0, 0, 0, 0, 'Shay Leafrunner - Actionlist - Set Flags Immune To Players & Immune To NPC\'s'),
(777403, 9, 2, 0, 0, 0, 100, 0, 0, 0, 0, 0, 0, 74, 61, 0, 0, 0, 0, 0, 1, 0, 0, 0, 0, 0, 0, 0, 0, 'Shay Leafrunner - Actionlist - Remove Timed Event 61');
```

Comment (Gultask) [reply data/sql/updates/pending_db_world/rev_1685747385631953200.sql:41]: Makes it more readable

Comment (The-GhostRider) [reply data/sql/updates/pending_db_world/rev_1685747385631953200.sql]: > Makes it more readable

Done

**Comment by Kitzunu** [comment]
Actually my bad, I was wrong.
I have learned that when a PR only touches smart scripts and nothing else creature related it should be DB/SAI

Sorry for that 😅

File: data/sql/updates/pending_db_world/rev_1685747385631953200.sql (hunk 1 above)
Comment (Gultask) [reply data/sql/updates/pending_db_world/rev_1685747385631953200.sql]: What I showed here was that I merged several INSERT queries into a single one, not that I removed the fields in the INSERT 😛

**Review by Gultask — COMMENTED** [review]
What's the reasoning for using IDs 61 and 38 for the Timed Events?

File: data/sql/updates/pending_db_world/rev_1685747385631953200.sql (hunk 1 above)
Comment (FrancescoBorzi) [reply data/sql/updates/pending_db_world/rev_1685747385631953200.sql]: @Gultask what's the status of this PR? is this review comment resolved? if yes, please mark it as resolved :)

**Comment by FrancescoBorzi** [comment]
@Gultask @Kitzunu what's the status of this PR? needs more test, changes or what? please don't forget to apply the workflow labels when reviewing PRs, so we avoid to forget PRs behind

**Comment by elthehablo** [comment]
> @Gultask @Kitzunu what's the status of this PR? needs more test, changes or what? please don't forget to apply the workflow labels when reviewing PRs, so we avoid to forget PRs behind

I will test it

**Comment by elthehablo** [comment]
I cannot review, buddy. I'm not a maintainer

**Review by FrancescoBorzi — APPROVED** [review]
approving on behalf of @elthehablo

**Pull request merged**